packages/opentelemetry-exporter-prometheus/src/prometheus.ts

```diff
@@ -143,18 +142,33 @@ export class PrometheusExporter implements MetricExporter {
     }
 
     if (metric instanceof Gauge) {
-      if (record.aggregator instanceof CounterSumAggregator) {
-        metric.set(labels, point.value as Sum);
-      } else if (record.aggregator instanceof ObserverAggregator) {
+      if (typeof point.value === 'number') {
+        if (
+          record.descriptor.metricKind === MetricKind.VALUE_OBSERVER ||
+          record.descriptor.metricKind === MetricKind.VALUE_RECORDER
+        ) {
+          metric.set(
+            labels,
+            point.value,
+            hrTimeToMilliseconds(point.timestamp)
+          );
+        } else {
+          metric.set(labels, point.value);
+        }
+      } else if ((point.value as Histogram).buckets) {
```

Comment:
What determines if a point value is a number, histogram, or distribution? Does this come from the aggregator? If so, what is the point of allowing multiple aggregators if sum will always be used. Shouldn't histograms be exported as histograms, not gauges?

Reply:
Previous approach is wrong. You can define your own batcher with different aggregators and then the same metric will have different aggregator and thus it will cause an error. So the value should be detected based on what aggregator it has not the metric kind.